File: Linguagem-C#/resolucao/resolucao_idades.cs
using System;
using System.Globalization;

namespace resolucao_idades
{
  class Program
  {
    static void Main(string[] args)
    {
      CultureInfo CI = CultureInfo.InvariantCulture;

      // declaracao das variaveis a serem utilizadas na resolucao //
      string nome1, nome2;
      int idade1, idade2;
      double media;

      // imput e organizacao da logica a ser utilizada para a resolucao //
      Console.WriteLine("Dados da primeira pessoa:");
      Console.Write("Nome: ");
      nome1 = Console.ReadLine();
      Console.Write("Idade: ");
      idade1 - int.Parse(Console.ReadLine());
      
      Console.WriteLine("Dados da segunda pessoa:");
      Console.Write("Nome: ");
      nome2 = Console.ReadLine();
      Console.Write("Idade: ");
      idade2 - int.Parse(Console.ReadLine());

      // calculo da media das idades solicitada //
      media = (double)(idade1 + idade2) / 2;

      // output da frase solicitada com o resultado da media das idades //
      Console.WriteLine("A idade media de " + nome1 + " e " + nome2 + " eh de " + media.ToString("F1", CI) + " anos");
    }
  }
}

File: Linguagem-C++/resolucoes_exercicios/resolucao-idades.cpp
#include <bits/stdc++.h>

using namespace std;

int main () 
{
    // declaracao das variaveis a serem utilizadas para a resolucao do problema //
    string nome1, nome2;
    int idade1, idade2;
    double media;

    // input para a coleta dos dados a serem utilizados na resolucao //
    cout << "Dados da primeira pessoa:" << endl;
    cout << "Nome: ";
    getline(cin, nome1);
    cout << "Idade: ";
    cin >> idade1;

    cout << "Dados da segunda pessoa:" << endl;
    cout << "Nome: ";
    cin.ignore(INT_MAX, '\n');
    getline (cin, nome2);
    cout << "Idade: ";
    cin >> idade2;

    // calculo da media das idades //
    media = (double) (idade1 + idade2) / 2;
    cout << "A idade media de " << nome1 << " e " << nome2 << " e de " << media << " anos" << endl;

    return 0;
}